src/lintingRpaas.ts

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { devOps, cli } from '@azure/avocado';
+import * as utils from './utils';
+import { getLinterResult } from "./momentOfTruth";
+import * as fs from "fs-extra";
+import * as YAML from "js-yaml";
+import { LintingResultParser, LintingResultMessage, Issue } from './momentOfTruthUtils'
+import { composeLintResult,Mutable } from './momentOfTruthPostProcessing';
+/**
+ * 1 run linter rpaas
+ * 2 check whether the reamde has openapi-subtype
+ */
+
+ function isRpaasBranch() {
+   const targetBranch = utils.getTargetBranch();
+   const RPaaSBranches = ["RPaaSMaster", "RPaaSDev"];
+   return RPaaSBranches.some((b) => b === targetBranch)
+ }
+ 
+ export class ReadmeParser {
+   readmeFile : string 
+   markDownContent: string
+   constructor(readmePath: string) {
+      this.readmeFile = readmePath
+      this.markDownContent = fs.readFileSync(this.readmeFile, "utf8");
+   }
+
+   public getGlobalConfigByName(Name:string) {
+       let rawMarkdown = this.markDownContent;
+       for (const codeBlock of utils.parseCodeblocks(rawMarkdown)) {
+         if (
+           !codeBlock.info ||
+           codeBlock.info.trim().toLocaleLowerCase() !== "yaml" ||
+           !codeBlock.literal
+         ) {
+           continue;
+         }
+         try {
+            const configs = YAML.safeLoad(codeBlock.literal) as any
+            if (configs && configs[Name]) {
+              return configs[Name];
+            }
+         }
+         catch(e) {
+            console.log(e)
+         }
+      }
+   }
+
+ }
+
+export class LintMsgTransformer {
+  constructor() {}
+
+  lintMsgToUnifiedMsg(msg: LintingResultMessage[]) {
+     const result = msg.map(
+      (it) => ({
+        type: "Result",
+        ...composeLintResult(it as unknown as Mutable<Issue>)
+      }))
+      return JSON.stringify(result)
+  }
+
+  rawErrorToUnifiedMsg(errType:string, errorMsg: string,config:string) {
+      const result = {
+        type: "Raw",
+        level: "Error",
+        message: errType,
+        time: new Date(),
+        extra: {
+          new: utils.targetHref(utils.getRelativeSwaggerPathToRepo(config)),
+          details: errorMsg,
+        },
+      };
+      return JSON.stringify(result);
+  }
+} 
+
+class UnifiedPipeLineStore {
```

Comment:
Can we have this extracted as common class to be used by other tool call? e.g. lintDiff call.